src/sammich/plugins/reminder.py

```diff
@@ -0,0 +1,113 @@
+import datetime
+import logging
+import re
+import time
+
+import requests
+from dotenv import dotenv_values
+from machine.plugins.base import MachineBasePlugin
+from machine.plugins.decorators import command
+
+secrets = dotenv_values(".secrets")
+
+SLACK_BOT_TOKEN = secrets["SLACK_BOT_TOKEN"]
+
+
+class ReminderPlugin(MachineBasePlugin):
+    def parse_command(self, command_text):
+        pattern = re.compile(r"\[(.*?)\]\s+\[(.+?)\]\s+\[(.+?)\]")
+        match = pattern.match(command_text)
+        if match:
+            channel_name, message, when = match.groups()
+            return channel_name, message, when
+        return None, None, None
+
+    def get_channel_id(self, channel_name):
+        # Use Slack API to get the channel ID from the channel name
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {SLACK_BOT_TOKEN}",
+        }
+        response = requests.get("https://slack.com/api/conversations.list", headers=headers)
```

Comment:
You could use https://dondebonair.github.io/slack-machine/api/#machine.plugins.base.MachineBasePlugin.web_client instead.

Reply:
The issues is the property it provides just provides us the names of channels and not their Id
![image](https://github.com/OWASP-BLT/BLT-Sammich/assets/108613416/2396e765-79a8-433c-8be8-e4a093f5d81f)

Reply:
It seems the `Channel` object should include the id too -- https://api.slack.com/methods/conversations.list#examples